addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -351,6 +395,13 @@ public static String getResourceAsString(String name) {
         return list;
     }
 
+    public AutomationParam getParam() {
+        if (param == null) {
```

Comment:
Can't we initialise param in constructor/init method?

Reply:
This is the standard pattern we use for GUI elements.
The advantage is that if we dont need the GUI elements (eg if we are in daemon / cmdline mode) then they wont get initialized.
This may even fail on systems without a GUI...

Reply:
oh ok, however can't we handle the GUI element loading at extension level? If we can do that, then we can avoid the null checks.